### ISSUE TITLE
Support/switch to SWC

### ISSUE THREAD

**Issue by 403-html** [cypress-io/cypress#27323]
### What would you like?

Recently, SWC has garnered substantial attention within the development community, and it has been making [remarkable](https://twitter.com/matthewbrookson/status/1681245247509082112) strides in terms of popularity and adoption. Its ability to optimize and transpile JavaScript code at an impressive speed, thanks to its utilization of WebAssembly, sets it apart from conventional pre-processors like Babel.

I would like to propose changing project to use [SWC](https://swc.rs/).

### Why is this needed?

Speeds up all building and running processes.

**Comment by nagash77** [comment]
Hi @403-html , thank you for submitting this feature request.  We do not currently have this lined up for work any time in the near future.  I just wanted to be upfront about this works priority.  For the time being I will route this to our product team for consideration.

**Comment by lmiller1990** [comment]
I agree with the concept of moving to something (anything) other than webpack, which showed us how great bundlers can be but is fairly dated and not nearly as fast as it's contemporaries.

I have more experience around esbuild - I would be interested in a proper investigation to see which of these would be a better candidate.

Some related issues:

- Simliar request (from me): https://github.com/cypress-io/cypress/issues/25928
- Some benchmarks if we use esbuild: https://github.com/cypress-io/cypress/issues/25533 

I don't think this will be part of core anytime soon, it would probably be a big breaking change - it would need to be opt in. You can have most of what you want by authoring a preprocessor - here's one for [esbuild](https://github.com/bahmutov/cypress-esbuild-preprocessor) that works great. 

The best thing to do right now would be create a swc preprocessor (one may already exist) and this would give insight into what would be involved / what kind of complexities you might run into. You can also use it straight away - no need to wait for the Cypress team to review anything.

If you are working on a swc preprocessor, please share it here - I can contribute or help out.

**Comment by TomasSlama** [comment]
BTW, you can use just [SWC loader](https://github.com/swc-project/swc-loader) in webpack so you don't have to rewrite the whole infrastructure.

**Comment by tbullardcollibra** [comment]
We're using `swc-loader` with our cypress webpack setup, but I'm having a hard time determining the equivalent of 
```
'@babel/plugin-transform-modules-commonjs',
    {
         loose: true,
    },
```
so cy.stub() isn't working as expected. If anyone reads this and has any input it's welcome!
